api/src/main/scala/com/lightbend/lagom/internal/api/Path.scala

```diff
@@ -7,57 +7,55 @@ import java.util.regex.Pattern
 
 import akka.util.ByteString
 import com.lightbend.lagom.javadsl.api.Descriptor.{ CallId, NamedCallId, PathCallId, RestCallId }
-import com.lightbend.lagom.javadsl.api.deser.RawId.PathParam
-import com.lightbend.lagom.javadsl.api.deser.{ RawId, RawIdDescriptor }
-import org.pcollections.{ PSequence, PMap, HashTreePMap, TreePVector }
 import play.utils.UriEncoding
 
-import scala.collection.JavaConverters._
 import scala.util.parsing.combinator.JavaTokenParsers
 
-case class Path(parts: Seq[PathPart], queryParams: Seq[String]) {
+case class Path(pathSpec: String, parts: Seq[PathPart], queryParams: Seq[String]) {
```

Comment:
It's about making it impossible for us to do the wrong thing. Yes, it's painful, but I personally find there is value in doing it. To make it slightly less painful you could create a package object with a type alias that overrides the one in Predef. But I'm not going to insist more than I already did if you are against it.